internal/objectstatus/objectstatus_test.go

```diff
@@ -129,3 +131,58 @@ func Test_ObjectStatus_Default(t *testing.T) {
 	expected := component.NodeStatusOK
 	assert.Equal(t, expected, os.Status())
 }
+
+func Test_getDeletedObjectStatus_StatusError(t *testing.T) {
```

Comment:
This part of the code can be tested as part of status method which is closer to the public method. Testing closer to the public method may also allow omission of tests outside of the package such as `pod_test.go`.